src/Pact/Analyze/Term.hs

```diff
@@ -1,54 +1,48 @@
-{-# LANGUAGE DeriveAnyClass     #-}
-{-# LANGUAGE DeriveDataTypeable #-}
-{-# LANGUAGE FlexibleInstances  #-}
-{-# LANGUAGE GADTs              #-}
-{-# LANGUAGE Rank2Types         #-}
-{-# LANGUAGE StandaloneDeriving #-}
-{-# LANGUAGE TypeOperators      #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveDataTypeable    #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE Rank2Types            #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+{-# LANGUAGE TypeOperators         #-}
 
 module Pact.Analyze.Term where
 
 import           Data.Data          (Data)
-import           Data.Map.Strict    (Map)
 import           Data.SBV           (HasKind, SymWord)
-import           Data.SBV.Control   (SMTValue)
 import           Data.Text          (Text)
+import           Prelude            hiding (Float)
 
+import           Pact.Analyze.Numerical
+import           Pact.Analyze.PrenexNormalize ()
 import           Pact.Analyze.Types
+import           Pact.Analyze.Util
 
 --
 -- TODO: merge this module with Pact.Analyze.Types once we are on GHC 8.2 and
```

Comment:
@bts I don't think this applies anymore. can you confirm?

Reply:
At this point we probably don't want to do precisely what this comment indicates -- merging the two modules -- but we need to have a better organizing principle around our `Types` situation. maybe we have a `Types` toplevel, and a few smaller modules underneath?

Reply:
At the end of the day, what we call `Term` is currently really another `Types` module. And the live at the same level and the relationship isn't exactly clear (at least to me)